core/src/test/scala/integration/kafka/admin/ReassignPartitionsIntegrationTest.scala

```diff
@@ -392,10 +391,36 @@ class ReassignPartitionsIntegrationTest extends ZooKeeperTestHarness {
         new TopicPartition("foo", 0) -> PartitionReassignmentState(Seq(0, 1, 3, 2), Seq(0, 1, 3), false)),
         true, Map(), false))
 
+    assignment
+  }
+
+  @Test
+  def testAtOrAboveMinISRCancellationSucceeds(): Unit = {
+    cluster = new ReassignPartitionsTestCluster(zkConnect)
+    val assignment = setupPartitionMovementBeforeCancellation(cluster)
+
     // shutdown one of the original replicas
     cluster.servers.find(_.config.brokerId == 2).get.shutdown()
 
-    // Cancel the reassignment should result in an exception
+    // Canceling the reassignment should be okay because
+    // there are enough replicas alive in the original list = {0, 1}
+    assertTrue(() => {
+      runCancelAssignment(cluster.adminClient, assignment, true)
+      true
+    })
+  }
+
+  @Test
+  def testBelowMinISRCancellationFails(): Unit = {
+    cluster = new ReassignPartitionsTestCluster(zkConnect)
+    val assignment = setupPartitionMovementBeforeCancellation(cluster)
+
+    // shutdown two of the original replicas. Cancellation would cause ISR to drop below minISR
+    cluster.servers.find(_.config.brokerId == 1).get.shutdown()
+    cluster.servers.find(_.config.brokerId == 2).get.shutdown()
+
+    // Cancel the reassignment should result in an exception because
+    // there are not enough alive replicas in the original list {0}
     assertThrows(classOf[TerseReassignmentFailureException], () => runCancelAssignment(cluster.adminClient, assignment, true))
```

Comment:
I don't see your new config in here, and with its default value of 1, shouldn't this test case fail?

Reply:
you are right. Updated the test.